client/src/components/map/legend/item-types/timeline/index.tsx

```diff
@@ -1,120 +1,205 @@
 'use client';
 
-import React, { useCallback, useMemo } from 'react';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 
 import { Root, Track, Thumb } from '@radix-ui/react-slider';
 import { PauseIcon, PlayIcon } from 'lucide-react';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
 
-import { cn } from '@/lib/classnames';
+import { layersSettingsAtom, timelineAtom } from '@/store';
 
 import { Button } from '@/components/ui/button';
 
 type LegendTypeTimelineProps = {
-  timeline: {
-    start: number;
-    end: number;
-    currentFrame: number;
-  };
-  onChangeCurrent: (year: number) => void;
-  onPlay: () => void;
-  isPlaying?: boolean;
-  className?: string;
+  description: string;
+  startYear: number;
```

Comment:
**DON'T CHANGE THIS**, _it's just an idea that we could do as a shareable component._

I can argue against the props you are sending or how you are managing this "generic" Timeline. It's true that right now it works, but it can't handle different frequencies, different type of date data. 

I think that an improvement will be to: 
- use Dates instead of years.
- use date-fns to translate all these params to something readable to the Slider 
- the interval could be an enum `day`, `week`, `month`.